pennylane/labs/dla/dense_util.py

```diff
@@ -234,6 +234,172 @@ def batched_pauli_decompose(H: TensorLike, tol: Optional[float] = None, pauli: b
+            com = qml.simplify(qml.commutator(oj, oi))
+            if not qml.equal(com, 0 * qml.Identity()):
+                return False
+
+        return True
+
+    if all(isinstance(op, np.ndarray) for op in ops):
+        for oi, oj in combinations(ops, 2):
+            com = oj @ oi - oi @ oj
+            if not np.allclose(com, np.zeros_like(com)):
+                return False
+
+        return True
+
+    return NotImplemented
+
+
+def check_cartan_decomp(k: List[PauliSentence], m: List[PauliSentence], verbose=True):
+    r"""Helper function to check the validity of a Cartan decomposition
+
+    Check whether of not the following properties are fulfilled.
+    .. math::
+
+        \begin{align}
+            [\mathfrak{k}, \mathfrak{k}] \subseteq \mathfrak{k} & \text{ (subalgebra)}\\
+            [\mathfrak{k}, \mathfrak{m}] \subseteq \mathfrak{m} & \text{ (reductive property)}\\
+            [\mathfrak{m}, \mathfrak{m}] \subseteq \mathfrak{k} & \text{ (symmetric property)}
+        \end{align}
+
+    Args:
+        k (List[PauliSentence]): List of operators of the vertical subspace
+        m (List[PauliSentence]): List of operators of the horizontal subspace
+        verbose: Flat to set whether failures to meet one of criteria should be printed
+
+    Returns:
+        bool: Whether or not all properties are fulfilled
+
+    .. see-also:: :func:`~cartan_decomposition`
+
+    **Example**
+
+    We first construct a simple Lie algebra.
+
+    >>> from pennylane import X, Z
+    >>> rom pennylane.labs.dla import concurrence_involution, even_odd_involution, cartan_decomposition
+    >>> generators = [X(0) @ X(1), Z(0), Z(1)]
+    >>> g = qml.lie_closure(generators)
+    >>> g
+    [X(0) @ X(1),
+     Z(0),
+     Z(1),
+     -1.0 * (Y(0) @ X(1)),
+     -1.0 * (X(0) @ Y(1)),
+     -1.0 * (Y(0) @ Y(1))]
+
+    We compute the Cartan decomposition with respect to the :func:`~concurrence_involution`.
+
+    >>> k, m = cartan_decomposition(g, concurrence_involution)
+    >>> k, m
+    ([-1.0 * (Y(0) @ X(1)), -1.0 * (X(0) @ Y(1))],
+     [X(0) @ X(1), Z(0), Z(1), -1.0 * (Y(0) @ Y(1))])
+
+    We can check the validity of the decomposition using ``check_cartan_decomp``.
+
+    >>> check_cartan_decomp(k, m)
+    True
+
+    """
+    if any(isinstance(op, np.ndarray) for op in k):
+        k = [qml.pauli_decompose(op).pauli_rep for op in k]
+    if any(isinstance(op, np.ndarray) for op in m):
+        m = [qml.pauli_decompose(op).pauli_rep for op in m]
+
+    if any(isinstance(op, Operator) for op in k):
+        k = [op.pauli_rep for op in k]
+    if any(isinstance(op, Operator) for op in m):
+        m = [op.pauli_rep for op in m]
+
+    k_space = qml.pauli.PauliVSpace(k, dtype=complex)
+    m_space = qml.pauli.PauliVSpace(m, dtype=complex)
+
+    # Commutation relations for Cartan pair
+    if not (check_kk := check_commutation(k, k, k_space)):
+        _ = print("[k, k] sub k not fulfilled") if verbose else None
```

Comment:
Is this to make it compact?

Reply:
I think this was to get around pylint complaining 😅

Reply:
Fair enough :D It's fine by me!